From 13aa1241c32f300fdf809fedcfd3f1974f54a022 Mon Sep 17 00:00:00 2001
From: Alexander Traud <pabstraud@compuserve.com>
Date: Wed, 6 Sep 2017 10:15:26 +0200
Subject: [PATCH] res_srtp: Add support for libsrtp2.1.

Asterisk is able to use libSRTP 2.0.x. However since libSRTP 2.1.x, the macro
SRTP_AES_ICM got renamed to SRTP_AES_ICM_128. Beside to still compile with
previous versions of libSRTP, this change allows libSRTP 2.1.x as well.

ASTERISK-27253 #close

Change-Id: I2e6eb3c3bc844fee8a624060a2eb6f182dc70315
---
 res/srtp/srtp_compat.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/res/srtp/srtp_compat.h b/res/srtp/srtp_compat.h
index 4ab39f318f..bf42082445 100644
--- a/res/srtp/srtp_compat.h
+++ b/res/srtp/srtp_compat.h
@@ -5,7 +5,11 @@
 
 #define crypto_policy_t srtp_crypto_policy_t
 
+#if defined(SRTP_AES_ICM_128)
+#define AES_128_ICM SRTP_AES_ICM_128
+#else
 #define AES_128_ICM SRTP_AES_ICM
+#endif
 #define HMAC_SHA1 SRTP_HMAC_SHA1
 
 #define err_status_t srtp_err_status_t
-- 
2.11.0

